cowbird/monitoring/monitoring.py

```diff
@@ -1,177 +1,103 @@
-
-    def start(self):
-        """
-        Start the monitoring so that events can be fired.
-        """
-        self.__event_observer.schedule(self,
-                                       self.__src_path,
-                                       recursive=self.__recursive)
-        self.__event_observer.start()
-
-    def stop(self):
-        """
-        Stop the monitoring so that events stop to be fired.
-        """
-        self.__event_observer.stop()
-        self.__event_observer.join()
-
-    def on_moved(self, event):
-        # type: (Union[DirMovedEvent, FileMovedEvent]) -> None
-        """
-        Called when a file or a directory is moved or renamed.
-
-        @param event: Event representing file/directory movement.
-        """
-        self.__callback.on_deleted(event.src_path)
-        # If moved outside of __src_path don't send a create event
-        if event.dest_path.startswith(self.__src_path):
-            # If move under subdirectory and recursive is False don't send a
-            # create event neither
-            if self.__recursive or \
-                    os.path.dirname(event.dest_path) == \
-                    os.path.dirname(self.__src_path):
-                self.__callback.on_created(event.dest_path)
-
-    def on_created(self, event):
-        # type: (Union[DirCreatedEvent, FileCreatedEvent]) -> None
-        """
-        Called when a file or directory is created.
-
-        @param event: Event representing file/directory creation.
-        """
-        self.__callback.on_created(event.src_path)
-
-    def on_deleted(self, event):
-        # type: (Union[DirDeletedEvent, FileDeletedEvent]) -> None
-        """
-        Called when a file or directory is deleted.
-
-        @param event: Event representing file/directory deletion.
-        """
-        self.__callback.on_deleted(event.src_path)
-
-    def on_modified(self, event):
-        # type: (Union[DirModifiedEvent, FileModifiedEvent]) -> None
-        """
-        Called when a file or directory is modified.
-
-        @param event: Event representing file/directory modification.
-        """
-        self.__callback.on_modified(event.src_path)
+LOGGER = get_logger(__name__)
 
 
-@six.add_metaclass(SingletonMeta)
-class Monitoring:
+class Monitoring(metaclass=SingletonMeta):
     """
     Class handling file system monitoring and registering listeners.
 
     .. todo:: At some point we will need a consistency function that goes through all monitored folder and make sure
               that monitoring services are up to date.
     """
 
-    def __init__(self):
+    def __init__(self, config=None):
+        # type: (AnySettingsContainer) -> None
+        """
+        Initialize the monitoring instance from configured application settings.
+
+        @param config: AnySettingsContainer object from which the db can be retrieved.
+                       The default value of None is only there to disable pylint E1120. The singleton instance
```

Comment:
I would favor a `pylint: disable=E1120` comment instead to avoid the error. 
Code behaviour is more important. 
Leave the `if not config` though in case an invalid None/empty dict is explicitly passed.

Reply:
In fact the E1120 are raised everywhere the singleton is instantiated using Monitoring(). I don't think adding the default None value cause much harm.